kitsune/users/tests/__init__.py

```diff
@@ -104,3 +105,30 @@ class Meta:
     jwt_id = "e19ed6c5-4816-4171-aa43-56ffe80dbda1"
     issued_at = "1565720808"
     profile = factory.SubFactory(ProfileFactory)
+
+
+class ConversationFactory:
+    """
+    Populate inboxes and outboxes with messages between a primary user and a number of other users.
+    """
+
+    def __init__(self, primary_user=None, number_of_other_users=2):
+        # Set or create the primary user and create some other users.
+        self.user = primary_user or UserFactory()
+        self.other_users = UserFactory.create_batch(number_of_other_users)
+        # Populate the inboxes and outboxes of the users.
+        for sender in self.other_users:
+            send_message([self.user], "foo", sender=sender)
+        send_message(self.other_users, "bar", sender=self.user)
+        # Confirm our expectations.
+        assert self.user.outbox.count() == 1
```

Comment:
Assertions should not be part of the factory. Factories should prepare the objects that we need for the testing and we can perform all assertions in the test itself